stl/inc/algorithm

```diff
@@ -4657,12 +4657,12 @@ _SampleIt _Sample_reservoir_unchecked(
 
 template <class _PopIt, class _SampleIt, class _Diff, class _RngFn>
 _SampleIt _Sample_selection_unchecked(
-    _PopIt _First, const _PopIt _Last, _Iter_diff_t<_PopIt> _PopSize, _SampleIt _Dest, _Diff _Count, _RngFn& _RngFunc) {
+    _PopIt _First, _Iter_diff_t<_PopIt> _Pop_size, _SampleIt _Dest, _Diff _Count, _RngFn& _RngFunc) {
     // source is forward *and* we know the source range size: use selection sampling (stable)
-    // pre: _PopIt is forward && _Count <= _PopSize
+    // pre: _PopIt is forward && _Count <= _Pop_size
     using _CT = common_type_t<_Iter_diff_t<_PopIt>, _Diff>;
-    for (; _Count > 0 && _First != _Last; ++_First, (void) --_PopSize) {
-        if (static_cast<_CT>(_RngFunc(_PopSize)) < static_cast<_CT>(_Count)) {
+    for (; _Pop_size; ++_First, (void) --_Pop_size) {

```

Comment:
We conventionally avoid boolean conversions in favor of explicit boolean expressions
```suggestion
    for (; _Pop_size > 0; ++_First, (void) --_Pop_size) {
```
(The suggested change is consistent with 4791.)

Reply:
Copy & Paste failed me. I think it is time to retire